site/content/contribute/webapp/end-to-end-tests.md

```diff
@@ -1,49 +1,62 @@
-4. Under /pages folder, review existing files to check whether the DOM structure already has an existing Page Object.
-    - If no file exists, add a new one containing the Page Object.
-    - If a file does exist, then update the corresponding page file accordingly.
-5. Under tests folder, review existing files to check whether a related tag or test already exists.
-    - If no file exists, add a new test file for both elements and action assertions.
-    - If a file does exist, then update the corresponding test file accordingly.
-6. Initiate E2E test commands, preferably with tags or partial testing for quick iteration and then full testing for final observation.
+The folder structure is based from [Cypress scaffold](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests.html#Folder-Structure) which was created on initial run.  Folders are:
+```
+|-- cypress
+  |-- fixtures
+  |-- integration
+  |-- support
+```  
+
+- `/cypress/fixtures` or [Fixture Files](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests.html#Fixture-Files):
+    1.  Fixtures are used as external pieces of static data that can be used by tests.
+    2.  Typically used with the `cy.fixture()` command and most often when stubbing Network Requests.
+- `/cypress/integration` or [Test Files](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests.html#Test-files)
+    1.  To start writing tests, 
+        - simply create a new file (e.g. `login_spec.js`) within `cypress/integration` folder and;
+        - refresh tests list in the Cypress Test Runner and a new file should have appeared in the list.
+- `/cypress/support` or [Support Files](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests.html#Support-file)
+    1.  The support file is a place to put reusable behaviour such as Custom Commands or global overrides that are wished to be applied and available to all of spec files.
+
+## What requires an E2E Test?
+
+1. All test cases defined in the Release Checklist
+2. New feature or story
+3. Bug fixes
+4. Cases that are not covered by unit or integration tests
+
+## Interested in Contributing to E2E Testing through Help Wanted Tickets
+
+1. All help wanted tickets are under [server repository's GitHub issues](https://github.com/mattermost/mattermost-server/issues?q=is%3Aissue+is%3Aopen+label%3A%22Up+For+Grabs%22). Look for issues with `Add E2E Tests` and `Up For Grabs` labels, and comment to let everyone know you're working on it.
+2. Each ticket is filled up with specific test steps and assertions that need to be accomplished as a minimum requirement.  Additional steps and assertions for robusts test implementation are much welcome.
+3. Join our channel at [UI Test Automation](https://community.mattermost.com/core/channels/ui-test-automation) and talk to us as fellow contributors, and collaborate and learn with one another.
+
+## Guide for Writing E2E Testing
```

Comment:
0/5 but is this guide detailed enough for someone to be able to pick up the ticket and start working on it? Have we passed it around/tested internally to verify?

Reply:
I'd say yes as a minimum but definitely there a lot to improve as we go along.
This is not tested internally.  Maybe one way to verify it is to distribute a UI automation ticket to other teams and see it from there.

Reply:
Probably a good idea. Maybe triage one ticket to XYZ and one to ABC with a link to the doc and explanation in the ticket comments?